### ISSUE TITLE
Sending messages unreliable

### ISSUE THREAD

**Issue by elektronaut69** [pimatic/homeduino#16]
Since the last update, sending is unreliable. There is a much longer delay between toggling the switch and the done message than before. The rf-switch reacts maybe once out of 5 times. There are no errors in the log:
```
debug [pimatic-homeduino]: data: "ACK"
```

**Comment by Icesory** [comment]
The problem is that the cheap receiver produce a lot of interrupts. The verification system tries to receive this and leave the state "Waiting".
@sweetpi added a listenBeforeTalk method, which waits until the state machine is in the end or waiting state.
https://github.com/pimatic/RFControl/blob/master/RFControl.cpp#L496
But this don´t work.
I think we can send while we are in the first verification state. This state don´t guarantee that we actually receive something.
Is the statemachine in verfication 2 or 3 we should wait until the receive is ready.

**Comment by elektronaut69** [comment]
So using another receiver would help? Which one would you suggest.

Ronny <notifications@github.com> schrieb am So., 1. Mär. 2015 17:37:

**Comment by sweetpi** [comment]
@elektronaut69 Could you test with the new version?

**Comment by elektronaut69** [comment]
I will, but not before tomorrow in the evening. I am on nightshift for the
next days.

sweet pi - sweet home <notifications@github.com> schrieb am So., 1. Mär.
2015 21:09:

**Comment by elektronaut69** [comment]
I tested the latest version. It works much better but still misses to toggle rfswitches sometimes. I had a look at the code. If I get it right you prioritize receiving over sending and if there is too much traffic, you discard the message to send after 5s. Shouldn't it be the other way round? If you automate some task it is most important, that the actuator reacts reliably. As we use 433 MHz devices most devices have no backchannel and most sensors do transmit data repeatedly so we do not loose important information if we miss to receive some messages.

**Comment by sweetpi** [comment]
It's the other way around. 

1. The message is never discarded. Just the delay is at max 5 secs to wait for no message in the air.
2. The new system should help that the message gets not send while some other sender is sending and should be delayed till the other sender has finished sending. It's really strange that you get worse results with the new version.

**Comment by elektronaut69** [comment]
OK. I understand. I'm not so good in understanding C++. Seems like the problem had to do with the hardware I was using. I rebuilt my setup twice today, first using the sender and receiver from my pilight installation and I did get the same bad reliability as before. After rebuilding everything again on another breadboard it is working now. Seems like the breadboard is not good. I have transferred all my rules to my pimatic-only setup and will test the next days if everything works. Will report back then.
